
From: Dmitry Torokhov <dtor_core@ameritech.net>

parkbd - zero-fill allocated serio structure to prevent Oops when
registering port.

Signed-off-by: Dmitry Torokhov <dtor@mail.ru>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/input/serio/parkbd.c |    1 +
 1 files changed, 1 insertion(+)

diff -puN drivers/input/serio/parkbd.c~fix-oops-in-parkbd drivers/input/serio/parkbd.c
--- 25/drivers/input/serio/parkbd.c~fix-oops-in-parkbd	2004-10-09 21:54:05.754122704 -0700
+++ 25-akpm/drivers/input/serio/parkbd.c	2004-10-09 21:54:05.757122248 -0700
@@ -160,6 +160,7 @@ static struct serio * __init parkbd_allo
 
 	serio = kmalloc(sizeof(struct serio), GFP_KERNEL);
 	if (serio) {
+		memset(serio, 0, sizeof(struct serio));
 		serio->type = parkbd_mode;
 		serio->write = parkbd_write,
 		strlcpy(serio->name, "PARKBD AT/XT keyboard adapter", sizeof(serio->name));
_
